src/raw/iter.rs

```diff
@@ -256,20 +267,69 @@ impl<'a> RawElement<'a> {
         })
     }
 
+    pub(crate) fn value_utf8_lossy(&self) -> Result<Option<Utf8LossyBson<'a>>> {
```

Comment:
Something like this could hypothetically become part of the public API, although we'd want to think about what it should return along the axes of (restricted enum, full enum) x (always owned, some kind of cow).

Personally, because parsing invalid utf8 is such a corner case, what I'd like to do for 3.0 is remove support for that from the Serde deserialization entirely and _just_ have it exposed via a method like this on the raw document iteration.  That would give an escape hatch for those few times it's needed and enable quite a bit of simplification in the deserializer.  If that sounds reasonable, I'll file a bug for that.

Reply:
Filed RUST-1998 for this.